flocker/control/_protocol.py

```diff
@@ -253,7 +315,19 @@ def __init__(self, agent):
         CommandLocator.__init__(self)
         self.agent = agent
 
+    @property
+    def logger(self):
+        """
+        ``with_eliot_context`` assumes that the decorated method's bound
+        instance has a logger, which it could, but for testing, it needs to use
+        the same logger as the supplied agent.
+
+        XXX This seems ugly.
+        """
+        return self.agent.logger
```

Comment:
Or just have a single module-level logger? I'm still not convinced (for Eliot in general, and here specifically) that one ever needs more than one.